aiida/manage/configuration/config.py

```diff
@@ -366,16 +369,44 @@ def get_option(self, option_name, scope=None, default=True):
         return value
 
     def store(self):
-        """Write the current config to file."""
-        if os.path.isfile(self.filepath):
-            self._backup(self.filepath)
+        """Write the current config to file.
+
+        .. note:: if the configuration file already exists on disk it will only be overwritten if there are changes
+            made to the configuration in memory. In that case, a backup of the original file on disk will be created.
```

Comment:
Is that the behavior we want? If I call "conf.store()" I expect the content to be written to disk (i.e. the timestamp of the config file should be updated).

I completely agree, of course, that a backup is only necessary when the content changes.

Reply:
Fair enough, I can take the actual write call out of the conditional